app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage_dict = {}
+
+    @wraps(func)
+    def wrapper(*args: tuple) -> Any:
+        if args in storage_dict:
+            print("Getting from cache")
+            return storage_dict[args]
+        else:
+            result = func(*args)
+            storage_dict[*args] = result
```

Comment:
```suggestion
            storage_dict[args] = result
```

Reply:
Hello! Could you please make a fedback, I've implemeted the new decision!

Reply:
Hello! Could you please make a fedback, I've implemeted the new decision!